### PR TITLE
feat: replace Pug by Lodash with cache

### LINKED ISSUE #423: Completely remove Pug (formerly Jade) dependency
Should be possible now that we have https://github.com/nodecg/nodecg/commit/94d0b1d791043c5453fb2a7edb7979ed13c5c327

### PR DESCRIPTION
**Pull request by Seldszar**
Closes #423

### CODE REVIEW

**Comment by Hoishin** [comment]
I don't know much about JavaScript template engines. Is lodash faster than pug?

**Comment by Seldszar** [comment]
I'm pretty sure they're both efficient. Pug converts the template into JavaScript code (https://github.com/pugjs/pug/blob/master/packages/pug-code-gen/index.js), like Lodash does.

**Comment by Lange** [comment]
The main reason I wanted to get away from Pug was that I wanted to use normal HTML syntax, which is what this PR achieves. I think there's room to use this kind of templating even less, and do more of it client-side, but this is a good start and solves the specific problem of getting rid of the special Pug syntax which I'd grown weary of dealing with.

**Comment by Hoishin** [comment]
Merging #439  revealed what fails the test 👏 
screen shot 2018-10-12 at 01 04 13

File: lib/login/public/login.tmpl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			<a href="/login/twitch">
+				<img src="/login/signin_twitch.png">
+			</a>
+		<% } %>
+
+		<% if (config.login.local && config.login.local.enabled) { %>
+			<form method="post" action="/login/local">
+				<div class="form-group">
+					<label for="username">Username:</label>
+					<input class="form-control" type="text" name="username" />
```
Comment (Hoishin) [inline lib/login/public/login.tmpl]: missing `id="username"`

File: lib/login/public/login.tmpl [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+				</div>
+				<div class="form-group">
+					<label for="password">Password:</label>
+					<input class="form-control" type="password" name="password" />
```
Comment (Hoishin) [inline lib/login/public/login.tmpl]: missing `id="password"`

**Review by Hoishin — CHANGES_REQUESTED** [review]

File: lib/login/public/login.tmpl (hunk 1 above)
Comment (Seldszar) [reply lib/login/public/login.tmpl]: Yes, I thought these IDs were useless.

**Review by Hoishin — APPROVED** [review]

File: lib/login/public/login.tmpl [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			<form id="localForm" method="post" action="/login/local">
+				<div class="form-group">
+					<label for="username">Username:</label>
+					<input id="username" class="form-control" type="text" name="username" />
+				</div>
+				<div class="form-group">
+					<label for="password">Password:</label>
+					<input id="password" class="form-control" type="password" name="password" />
+				</div>
+				<button class="btn btn-primary" type="submit">Login</button>
```
Comment (Hoishin) [inline lib/login/public/login.tmpl]: Also missing `id="localSubmit"` here

**Review by Hoishin — CHANGES_REQUESTED** [review]

**Review by Hoishin — APPROVED** [review]

**Pull request merged**